ProcessMaker/Models/Script.php

```diff
@@ -143,7 +150,29 @@ public function runScript(array $data, array $config, $tokenId = '', $errorHandl
             throw new \RuntimeException('A user is required to run scripts');
         }
 
-        return $runner->run($this->code, $data, $config, $this->timeout(), $user);
+        try {
+            return $runner->run($this->code, $data, $config, $this->timeout(), $user);
+        } catch (\RuntimeException $e) {
+            Log::error($e);
+            if ($this->retryAttempts() !== null && $this->retryWaitTime() !== null) {
+                Log::error('Retry the runScript process.');
+                if ($this->attemptedRetries >= $this->retryAttempts()) {
+                    $message = "Script failed after $this->attemptedRetries attempts";
+                    Log::error($message);
+                    throw new \RuntimeException($message);
```

Comment:
Lets append the original message so the user can see what caused it `$message . "\n" . $e->getMessage()`